### DISCUSSION TITLE
Next.js adds unnecessary  data-next-hide-fouc when handling route from custom server

### DISCUSSION THREAD

**Discussion by ivorpad** [vercel/next.js#16104] [Help]
I have a custom Koa server that passes the route handling to Next if a route has a specific path, but the page has `data-next-hide-fouc` with `body {display: none}` added to `<head>` — I understand this is to [prevent FOUC](https://github.com/vercel/next.js/issues/10285), but how do I handle this situation without adding an `!important` declaration?

I believe this is a bug.

```js
server.use(graphQLProxy.unless({path: "/custom_route"}));
server.use(verifyRequest.unless({ path: "/custom_route" }));
server.use(createShopifyAuth.unless({ path: "/custom_route" }));
```

```javascript
router.get("/custom_route", async (ctx) => {
  await handle(ctx.req, ctx.res);
  ctx.respond = false;
  ctx.res.statusCode = 200;
});
```

<img width="556" alt="Screen Shot 2020-08-12 at 9 52 13 AM" src="https://user-images.githubusercontent.com/52452/89989662-8c3e3380-dc81-11ea-8d4b-1bb1c59ff901.png">

**Comment by timneutkens** [comment]
The fouc styles are only there in development to prevent a FOUC from styles being injected by JS (for hot reloading purposes), it is automatically removed by Next.js.

Please follow the issue template when creating issues, converted this to a discussion.

**Reply by ivorpad** [reply]
Hey @timneutkens,

I understand why is there, but how am I supposed to style the page with the body hidden? As you read above, this appear when I let Next.js handle the route from a custom server.

**Reply by ivorpad** [reply]
Added an ugly style tag in `_document.js` 

IMHO:
1. This is a bug
2. We should have a flag to disable it in dev mode.

```css
<style>{`body { display: block !important }`}</style>
```

**Reply by timneutkens** [reply]
As said:

> it is automatically removed by Next.js.

You likely have something wrong in your setup, potentially JS is not being served correctly or something like that.

**Reply by meandillar** [reply]
We're just setting up an embedded Shopify app with a custom server and came across the same bug. Using next version 11.1.0. Had to add the style tag to make it work. Definitely seems like a bug.

**Comment by shanbady** [comment]
This is due to your adblocker. disable adblock

**Reply by neoneon28** [reply]
This was exactly the problem, not sure why so many downvotes.

This helped me thank you :)

**Comment by Izhaki** [comment]
I had a related issue with NextJS not showing up in Cypress.

Turns out hijacking the clock was the reason:

https://github.com/vercel/next.js/issues/10285#issuecomment-820359832

**Comment by sanathsharma** [comment]
Had the same issue when using custom server. Was able to solve it by handling `GET` request to `_next*` route. There were lot of 404 errors in the console related to that.

**Comment by dalvallana** [comment]
I am following this example from NextJS: https://nextjs.org/docs/messages/returning-response-body-in-middleware.

I want to show a 403 Unauthorized page. But I don't know why the body's `display: none` isn't removed as with other pages.

**Comment by chriskrogh** [comment]
for me, the issue was that i was running my custom node server using [ts-node-dev](https://www.npmjs.com/package/ts-node-dev). once i switched back over to `ts-node`, `display: none` was no longer present on `body`.

**Comment by Sanusihassan** [comment]
One possible solution to the issue you mentioned could be the presence of a `<base>` tag in the HTML document. The `<base>` tag specifies the base URL for all relative URLs in an HTML document, allowing the developer to set a default URL that will be used to resolve any relative URLs that appear in the document. If there is a `<base>` tag present in the document, it could potentially change the URLs in the browser and affect how relative URLs are resolved. It might be worth checking if there is a `<base>` tag present in the document and, if so, whether it is affecting the URLs in the browser.
